packages/e2e-tests/specs/experiments/blocks/comments-query.test.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	activateTheme,
+	createNewPost,
+	insertBlock,
+	pressKeyTimes,
+	publishPost,
+	setOption,
+	trashAllComments,
+} from '@wordpress/e2e-test-utils';
+
+describe( 'Comment Query Loop', () => {
+	let previousPageComments,
+		previousCommentsPerPage,
+		previousDefaultCommentsPage;
+	beforeAll( async () => {
+		await activateTheme( 'emptytheme' );
+		previousPageComments = await setOption( 'page_comments', '1' );
+		previousCommentsPerPage = await setOption( 'comments_per_page', '1' );
+		previousDefaultCommentsPage = await setOption(
+			'default_comments_page',
+			'newest'
+		);
+	} );
+	beforeEach( async () => {
+		await createNewPost();
+	} );
+	it( 'Pagination links are working as expected', async () => {
+		// Insert the Query Comment Loop block.
+		await insertBlock( 'Comments Query Loop' );
+		// Insert the Comment Loop form.
+		await insertBlock( 'Post Comments Form' );
+		await publishPost();
+		// Visit the post that was just published.
+		await page.click(
+			'.post-publish-panel__postpublish-buttons .is-primary'
+		);
+
+		// TODO: We can extract this into a util once we find we need it elsewhere.
+		// Create three comments for that post.
+		for ( let i = 0; i < 3; i++ ) {
+			await page.waitForSelector( 'textarea#comment' );
+			await page.click( 'textarea#comment' );
+			await page.type(
+				`textarea#comment`,
+				`This is an automated comment - ${ i }`
+			);
+			await pressKeyTimes( 'Tab', 1 );
+			await page.keyboard.press( 'Enter' );
+			await page.waitForNavigation();
+		}
+
+		// We check that there is a previous comments page link.
+		await page.waitForSelector( '.wp-block-comments-pagination-previous' );
```

Comment:
There's one minor issue with this: If the selector is not there, this will time out and throw an error -- i.e. we won't even make it to the assertion on the next line. This means that the assertion is kind of redundant. 

So should we remove the assertion? No, since it's better to explicitly state our intent rather than implicitly.

Let's keep this as-is for now and merge it; I might file a follow-up to experiment with replacing the `waitForSelector` with `waitForNavigation` 🙂

Reply:
Tentative diff for later:

```diff
diff --git a/packages/e2e-tests/specs/experiments/blocks/comments-query.test.js b/packages/e2e-tests/specs/experiments/blocks/comments-query.test.js
index 55c6b4b07a..d3b8c880a9 100644
--- a/packages/e2e-tests/specs/experiments/blocks/comments-query.test.js
+++ b/packages/e2e-tests/specs/experiments/blocks/comments-query.test.js
@@ -48,12 +48,13 @@ describe( 'Comment Query Loop', () => {
 				`This is an automated comment - ${ i }`
 			);
 			await pressKeyTimes( 'Tab', 1 );
-			await page.keyboard.press( 'Enter' );
-			await page.waitForNavigation();
+			await Promise.all( [
+				page.keyboard.press( 'Enter' ),
+				page.waitForNavigation( { waitUntil: 'networkidle0' } ),
+			] );
 		}
 
 		// We check that there is a previous comments page link.
-		await page.waitForSelector( '.wp-block-comments-pagination-previous' );
 		expect(
 			await page.$( '.wp-block-comments-pagination-previous' )
 		).not.toBeNull();
@@ -61,21 +62,25 @@ describe( 'Comment Query Loop', () => {
 			await page.$( '.wp-block-comments-pagination-next' )
 		).toBeNull();
 
-		await page.click( '.wp-block-comments-pagination-previous' );
+		await Promise.all( [
+			page.click( '.wp-block-comments-pagination-previous' ),
+			page.waitForNavigation( { waitUntil: 'networkidle0' } ),
+		] );
 
 		// We check that there are a previous and a next link.
-		await page.waitForSelector( '.wp-block-comments-pagination-previous' );
-		await page.waitForSelector( '.wp-block-comments-pagination-next' );
 		expect(
 			await page.$( '.wp-block-comments-pagination-previous' )
 		).not.toBeNull();
 		expect(
 			await page.$( '.wp-block-comments-pagination-next' )
 		).not.toBeNull();
-		await page.click( '.wp-block-comments-pagination-previous' );
+
+		await Promise.all( [
+			page.click( '.wp-block-comments-pagination-previous' ),
+			page.waitForNavigation( { waitUntil: 'networkidle0' } ),
+		] );
 
 		// We check that there is only have a next link
-		await page.waitForSelector( '.wp-block-comments-pagination-next' );
 		expect(
 			await page.$( '.wp-block-comments-pagination-previous' )
 		).toBeNull();

```

Reply:
PR: https://github.com/WordPress/gutenberg/pull/39818